### PR TITLE
Issue 01 GitHub actions

### LINKED ISSUE #1: Set up GitHub actions
- [x] Set up unit tests for different python versions.
- [x] Set up unit tests for different operating systems.
- [x] Set up unit tests coverage test.
- [ ] Set up sphinx documentation test.
- [x] Set up copyright tests.
- [x] Set up flake8 tests.

### PR DESCRIPTION
**Pull request by siting-miao**
Closes #1

### CODE REVIEW

File: .github/workflows/coverage-test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install .
+          python -m pip install coverage codecov
+      - name: run coverage
+        run: |
+          coverage run run-tests.py --unit
+      - name: codecov
+        if: success()
+        env:
+          CODECOV_TOKEN: ${{ f4a14d78-425e-41b2-b28e-0a6d2fe886d7 }}
```
Comment (DavAug) [inline .github/workflows/coverage-test.yml:38]: Could you please generate a new codecov token, store it in the github secrets and then refer to the token here, so the token is not publicly visible? Let me know if you need help with that.

**Review by DavAug — CHANGES_REQUESTED** [review]
This looks really good @siting-miao !

The flake 8 and copyright tests are still failing, so could you look at the errors and fix the appropriate files.

Could you also add badges for the unit tests (python and os versions) and the coverage to the readme. you can again checkout pints for help.

File: .github/workflows/coverage-test.yml (hunk 1 above)
Comment (siting-miao) [reply .github/workflows/coverage-test.yml:38]: OK! Thank you. I’ll make those changes and ask you for help if I can’t😁.

**Comment by DavAug** [comment]
Hi @siting-miao 
this looks really good!
I've just created a new branch called issue-06-SEIR-model, so we can work on the implementation of the model separate from the github actions.

**Comment by DavAug** [comment]
This would just require us to revert a couple of changes that have to do with the SEIR model for this branch, and add them to the main branch through the branch issue-06-SEIR-model. I hope that makes sense.

File: seirmo/SEIRModelClass.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+"""#
```
Comment (DavAug) [inline seirmo/SEIRModelClass.py:1]: Could you please delete this file. It is saved in the branch issue-06-SEIR-model, so we can merge the github actions.

File: seirmo/__init__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+"""#
+# This file is part of seirmo (https://github.com/SABS-R3-Epidemiology/seirmo/)
+# which is released under the BSD 3-clause license. See accompanying LICENSE.md
+# for copyright notice and full license details.
+#"""
```
Comment (DavAug) [inline seirmo/__init__.py:5]: We forgot to do this with @FarmHJ PR.

Could you please register the FowardModel class by importing it like this:
```
from .models import (
   ForwardModel
)

```

File: seirmo/dummy.py [hunk 4]
```diff
@@ -1,7 +1,13 @@
+"""#
```
Comment (DavAug) [inline seirmo/dummy.py:1]: We can delete this file now because we'll have tests for the ForwardModel.

File: seirmo/models.py [hunk 5]
```diff
@@ -1,7 +1,13 @@
+"""#
+# This file is part of seirmo (https://github.com/SABS-R3-Epidemiology/seirmo/)
+# which is released under the BSD 3-clause license. See accompanying LICENSE.md
+# for copyright notice and full license details.
+#"""
 #
```
Comment (DavAug) [inline seirmo/models.py:6]: Could you possible remove the comment `# Forward model` it's not really necessary. The license header is enough, i.e.

```
#
# This file is part of seirmo (https://github.com/SABS-R3-Epidemiology/seirmo/)
# which is released under the BSD 3-clause license. See accompanying LICENSE.md
# for copyright notice and full license details.
# 
```
Could you also please remove the """ at the beginning and end of the header?

File: seirmo/tests/test_dummy.py [hunk 6]
```diff
@@ -1,3 +1,8 @@
+"""#
```
Comment (DavAug) [inline seirmo/tests/test_dummy.py:1]: This file can also be deleted.

File: seirmo/version_info.py [hunk 7]
```diff
@@ -1,3 +1,8 @@
+"""#
```
Comment (DavAug) [inline seirmo/version_info.py:1]: Could you remove the wrapping """ here too?

File: setup.py [hunk 8]
```diff
@@ -60,6 +60,8 @@ def get_readme():
     install_requires=[
         # Dependencies go here!
         'numpy>=1.8',
+        'matplotlib',
```
Comment (DavAug) [inline setup.py:63]: I don't think we need the matplotlib dependency, do we? So could you please remove it?

**Comment by siting-miao** [comment]
@DavAug Yes, that makes perfect sense! :) please give me some comments on the SEIRModel, I'll continue that tomorrow when I get up (probably at noon haha).

**Review by DavAug — CHANGES_REQUESTED** [review]
This looks really raelly good @siting-miao !

There are just a couple minor comments. We forgot to test the ForwardModel class before merging it, so could you please add a file `test_models.py` to the tests directory and basically copy what the branchpro guys have done, https://github.com/SABS-R3-Epidemiology/branchpro/blob/main/branchpro/tests/test_models.py. You would just have to replace the branchpro by seirmo.

**Comment by DavAug** [comment]
> @DavAug Yes, that makes perfect sense! :) please give me some comments on the SEIRModel, I'll continue that tomorrow when I get up (probably at noon haha).

Ok, I'll make some comments on the implementation of the SEIRModel in the associated issue #6 .

**Comment by siting-miao** [comment]
> > @DavAug Yes, that makes perfect sense! :) please give me some comments on the SEIRModel, I'll continue that tomorrow when I get up (probably at noon haha).
> 
> Ok, I'll make some comments on the implementation of the SEIRModel in the associated issue #6 .

Thx! And I'll do those modifications you've suggested now.

File: seirmo/models.py [hunk 9]
```diff
@@ -1,7 +1,13 @@
...
 # ForwardModel Class
 #
 
+
```
Comment (DavAug) [inline seirmo/models.py:10]: Could you also please add a `self` to the simulate method? Seems like I've missed quite a few slips when reviewing the other PR :D sorry!

**Comment by siting-miao** [comment]
No worries I’ll do that 😉

________________________________

File: seirmo/__init__.py [hunk 10]
```diff
@@ -0,0 +1,9 @@
+#
+# This file is part of seirmo (https://github.com/SABS-R3-Epidemiology/seirmo/)
+# which is released under the BSD 3-clause license. See accompanying LICENSE.md
+# for copyright notice and full license details.
+#
+
+from .models import (
+    ForwardModel
+)
```
Comment (DavAug) [inline seirmo/__init__.py:9]: blank line

**Review by DavAug — APPROVED** [review]

**Pull request merged**